### DISCUSSION TITLE
proxy with supabase client js?

### DISCUSSION THREAD

**Discussion by pekado** [supabase/supabase#2385] [Questions] [answered]
Hi!

I'm working with supabase client and I need to hide the url, I already have a proxy in my webpack configuration. Is there a way to change the base url from createClient method or I have to start using fetch? 

If I start using fetch, where is the API for access token verification?

My proxy:
` devServer: {
    contentBase: path.join(__dirname, '/public'),
    port: 8000,
    historyApiFallback: true,
    proxy: {
      '/api/*': {
        target: 'SUPABASE_RESTURL',
        pathRewrite: { '^/api': '' },
        secure: false,
        changeOrigin: true,
        headers: {
          'Access-Control-Allow-Origin': '*',
        },
      },
    },
  },`

Thanks!

**Comment by inian** [comment, marked as answer]
May I know the use-case behind changing the base URL @pekado? supabase-js uses [postgrest-js](https://github.com/supabase/postgrest-js) and if you use that directly, you have full control over the base URL.

**Reply by pekado** [reply]
Security, the proxy hides the server url on the network tab.

**Comment by CMCDragonkai** [comment]
I'd like to be able to send requests to a proxy for debugging the requests. Is this possible?

**Comment by shafouz** [comment]
debugging with proxy setup:

```js
// npm i global-agent --save-dev
// GLOBAL_AGENT_HTTPS_PROXY=http://127.0.0.1:8080 GLOBAL_AGENT_HTTP_PROXY=http://127.0.0.1:8080 node -r global-agent/bootstrap ./lab.js

import { createClient } from '@supabase/supabase-js'
import fetch from 'node-fetch';

const options = {
  global: {
    fetch: fetch.bind(globalThis)
  },
}

const supabase = createClient('http://localhost:8000', '<jwt>', options)

const { data, error } = await supabase
  .from('ips')
  .select()
```
